src/test/java/gov/healthit/chpl/aqa/asserts/ListingHistoryEyeAsserts.java

```diff
@@ -0,0 +1,64 @@
+package gov.healthit.chpl.aqa.asserts;
+
+import static org.junit.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import cucumber.api.java.en.Then;
+import gov.healthit.chpl.aqa.model.ListingHistoryData;
+import gov.healthit.chpl.aqa.pageObjects.ListingHistoryEyePage;
+import gov.healthit.chpl.aqa.stepDefinitions.Base;
+
+/**
+ * Class ListingHistoryEyeAsserts definition.
+ */
+public class ListingHistoryEyeAsserts extends Base {
+
+    /**
+     * Assert that expected product edit details is displayed on product history eye.
+     * @param expectedEditDetails is "Developer/Product/Version change"
+     * @throws Exception of type ParseException or InterruptedException
+     */
+    @Then("^I see the listing history eye shows a \"([^\"]*)\" on today's date$")
+    public void iSeeListingHistory(final String expectedEditDetails) throws Exception {
+        Thread.sleep(SLEEP_TIME);
+        String actualListingHistoryDetails = ListingHistoryEyePage.listingHistoryTable(getDriver()).getText();
+        String[] listingHistoryDetails = actualListingHistoryDetails.split("\n");
+        int activityCount = 0;
+        ListingHistoryData listingHistoryDataInst = null;
+        String date = null;
+        for (String activity : listingHistoryDetails) {
+            if ((activityCount % 2) == 0) {
+                date = activity;
+            } else {
+                if (activity.startsWith(expectedEditDetails)) {
+                    listingHistoryDataInst = new ListingHistoryData();
+                    listingHistoryDataInst.setDate(date);
+                    listingHistoryDataInst.setActivity(activity);
+                    break;
```

Comment:
The way I read this, the loop exits as soon as it finds any activity that starts with the "expectedEditDetails" value. What happens if the table is sorted backwards?

Reply:
In this case, I'm assuming the first row is the latest updated one. So I'm always considering the first row for which the activity is matched.

Reply:
I recognize that, and that's exactly the problem. What happens if some other process updates the same Listing in-between your update and your check? What happens if, for whatever reason, we decide to change the order of these items? What happens if we end up adding a new activity type that is added when you change the name, and it shows up as more recent than the name change? This code _assumes_ the most recent item is the right one, but it doesn't need to, and by doing so it makes the test more brittle

Reply:
I understand, would try to come up with a better approach. Thanks!